### PR TITLE
feat(api): add otp (sms and email) checks in session api

### PR DESCRIPTION
**Pull request by livio-a**
This PR implements One-Time Password (OTP) with email and SMS on the v2alpha session API.

closes [#6224](https://github.com/zitadel/zitadel/issues/6224)

### Definition of Ready

- [x] I am happy with the code
- [x] Short description of the feature/issue is added in the pr description
- [x] PR is linked to the corresponding user story
- [x] Acceptance criteria are met
- [x] All open todos and follow ups are defined in a new ticket and justified
- [x] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [x] No debug or dead code
- [x] My code has no repetitions
- [x] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [x] All non-functional requirements are met
- [x] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

File: internal/command/session_otp.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+	return c.createOTPSMSChallenge(true, dst)
+}
+
+func (c *Commands) CreateOTPSMSChallenge() SessionCommand {
+	return c.createOTPSMSChallenge(false, nil)
+}
+
+func (c *Commands) createOTPSMSChallenge(returnCode bool, dst *string) SessionCommand {
+	return func(ctx context.Context, cmd *SessionCommands) error {
+		writeModel := NewHumanOTPSMSWriteModel(cmd.sessionWriteModel.UserID, "")
```
Comment (stebenz) [inline internal/command/session_otp.go:24]: Do we need a check that the session already contains the userID?

File: internal/command/session_otp.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+	return c.createOTPEmailChallenge(true, "", dst)
+}
+
+func (c *Commands) CreateOTPEmailChallenge() SessionCommand {
+	return c.createOTPEmailChallenge(false, "", nil)
+}
+
+func (c *Commands) createOTPEmailChallenge(returnCode bool, urlTmpl string, dst *string) SessionCommand {
+	return func(ctx context.Context, cmd *SessionCommands) error {
+		writeModel := NewHumanOTPEmailWriteModel(cmd.sessionWriteModel.UserID, "")
```
Comment (stebenz) [inline internal/command/session_otp.go:82]: Same here, check if userID already != ""?

File: internal/command/session_otp_test.go [hunk 3]
```diff
@@ -0,0 +1,856 @@
...
+		err        error
+		returnCode string
+		commands   []eventstore.Command
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		res    res
+	}{
+		{
```
Comment (stebenz) [inline internal/command/session_otp_test.go:34]: If userID check added, tests should also be added.

File: internal/command/session_otp_test.go [hunk 4]
```diff
@@ -0,0 +1,856 @@
...
+		err      error
+		commands []eventstore.Command
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		res    res
+	}{
+		{
+			name: "otp not ready, precondition error",
```
Comment (stebenz) [inline internal/command/session_otp_test.go:117]: If userID check added, tests should also be added.

File: internal/command/session_otp_test.go [hunk 5]
```diff
@@ -0,0 +1,856 @@
...
+		sessionID     string
+		resourceOwner string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr error
+	}{
+		{
```
Comment (stebenz) [inline internal/command/session_otp_test.go:197]: If userID check added, tests should also be added.

File: internal/command/session_otp_test.go [hunk 6]
```diff
@@ -0,0 +1,856 @@
...
+		err           error
+		commands      []eventstore.Command
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		res    res
+	}{
+		{
```
Comment (stebenz) [inline internal/command/session_otp_test.go:274]: If userID check added, tests should also be added.

File: internal/command/session_otp_test.go [hunk 7]
```diff
@@ -0,0 +1,856 @@
...
+		err        error
+		returnCode string
+		commands   []eventstore.Command
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		res    res
+	}{
+		{
```
Comment (stebenz) [inline internal/command/session_otp_test.go:378]: If userID check added, tests should also be added.

File: internal/command/session_otp_test.go [hunk 8]
```diff
@@ -0,0 +1,856 @@
...
+	type res struct {
+		err      error
+		commands []eventstore.Command
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		res    res
+	}{
+		{
```
Comment (stebenz) [inline internal/command/session_otp_test.go:461]: If userID check added, tests should also be added.

**Comment by livio-a** [comment]
@stebenz you're right about the userID check... i've added them and the tests

**Review by stebenz — APPROVED** [review]

**Pull request merged**